### ISSUE TITLE
Encountered an issue when compiling the 3.3.5 on centos6

### ISSUE THREAD

**Issue by wplatform** [TrinityCore/TrinityCore#24021]
In file included from /home/mywow/TrinityCore/dep/jemalloc/include/jemalloc/internal/jemalloc_preamble.h:48:0,
                 from /home/mywow/TrinityCore/dep/jemalloc/src/jemalloc.c:2:
/home/mywow/TrinityCore/dep/jemalloc/src/jemalloc.c: In function ‘jemalloc_secure_getenv’:
/home/mywow/TrinityCore/dep/jemalloc/include/jemalloc/internal/test_hooks.h:15:37: error: ‘secure_getenv’ undeclared (first use in this function)
 #define secure_getenv JEMALLOC_HOOK(secure_getenv, test_hooks_libc_hook)
                                     ^
/home/mywow/TrinityCore/dep/jemalloc/include/jemalloc/internal/test_hooks.h:7:71: note: in definition of macro ‘JEMALLOC_HOOK’
 #define JEMALLOC_HOOK(fn, hook) ((void)(hook != NULL && (hook(), 0)), fn)
                                                                       ^~
/home/mywow/TrinityCore/dep/jemalloc/src/jemalloc.c:714:9: note: in expansion of macro ‘secure_getenv’
  return secure_getenv(name);
         ^~~~~~~~~~~~~
/home/mywow/TrinityCore/dep/jemalloc/include/jemalloc/internal/test_hooks.h:15:37: note: each undeclared identifier is reported only once for each function it appears in
 #define secure_getenv JEMALLOC_HOOK(secure_getenv, test_hooks_libc_hook)
                                     ^
/home/mywow/TrinityCore/dep/jemalloc/include/jemalloc/internal/test_hooks.h:7:71: note: in definition of macro ‘JEMALLOC_HOOK’
 #define JEMALLOC_HOOK(fn, hook) ((void)(hook != NULL && (hook(), 0)), fn)
                                                                       ^~
/home/mywow/TrinityCore/dep/jemalloc/src/jemalloc.c:714:9: note: in expansion of macro ‘secure_getenv’
  return secure_getenv(name);
         ^~~~~~~~~~~~~
make[2]: *** [dep/jemalloc/CMakeFiles/jemalloc.dir/src/jemalloc.c.o] Error 1
make[1]: *** [dep/jemalloc/CMakeFiles/jemalloc.dir/all] Error 2
make: *** [all] Error 2

**Comment by Aokromes** [comment]
centos (specialy 6) is not recommended to use to compile tc.
https://trinitycore.atlassian.net/wiki/spaces/tc/pages/10977288/Linux+Requirements

**Comment by jackpoz** [comment]
You could disable jemalloc with the cmake option -DNOJEM=TRUE

**Comment by jackpoz** [comment]
It seems to be an open issue at jemalloc https://github.com/jemalloc/jemalloc/issues/1082

**Comment by wplatform** [comment]
Hi @Aokromes,  Thanks for answering this question. Centos is the most popular platform to run the backend application. I want to know the reason why Centos is not recommended to use to compile, and are there workarounds(I don't want to change my os, but I can upgrade to 7.x, Dose centos 7 works?).

I put the CMake option '-DNOJEM=1' the command. It can work but encounter another issue when installing. I am not a c++ developer. I need help to fix these issues. Thanks.

[  6%] Built target revision_data.h
[  6%] Building CXX object src/common/CMakeFiles/common.dir/IPLocation/IPLocation.cpp.o
In file included from /home/mywow/TrinityCore/src/common/PrecompiledHeaders/commonPCH.h:24:0,
                 from /home/mywow/TrinityCore/build/src/common/cotire/common_CXX_prefix.cxx:4,
                 from /home/mywow/TrinityCore/build/src/common/cotire/common_CXX_prefix.hxx:4:
/home/mywow/TrinityCore/src/common/IPLocation/IPLocation.cpp: In member function ‘void IpLocationStore::Load()’:
/home/mywow/TrinityCore/src/common/Define.h:137:20: error: expected ‘)’ before ‘PRIuPTR’
 #define SZFMTD "%" PRIuPTR
                    ^
/home/mywow/TrinityCore/src/common/Logging/Log.h:159:32: note: in definition of macro ‘TC_LOG_MESSAGE_BODY’
                     check_args(__VA_ARGS__);                            \
                                ^~~~~~~~~~~
/home/mywow/TrinityCore/src/common/IPLocation/IPLocation.cpp:97:5: note: in expansion of macro ‘TC_LOG_INFO’
     TC_LOG_INFO("server.loading", ">> Loaded " SZFMTD " ip location entries.", _ipLocationStore.size());
     ^~~~~~~~~~~
/home/mywow/TrinityCore/src/common/IPLocation/IPLocation.cpp:97:48: note: in expansion of macro ‘SZFMTD’
     TC_LOG_INFO("server.loading", ">> Loaded " SZFMTD " ip location entries.", _ipLocationStore.size());
                                                ^~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [src/common/CMakeFiles/common.dir/IPLocation/IPLocation.cpp.o] Error 1
make[1]: *** [src/common/CMakeFiles/common.dir/all] Error 2

**Comment by Shauren** [comment]
Centos is not recommended specifically because it very often uses old versions of everything and it does weird things to its libraries and header files that no other distro does causing issues like this one

**Comment by wplatform** [comment]
Hi @Shauren, Are there workarounds, Can I download the release package like 'TDB_full_world_335.19121_2019_12_15.7z' to install directly rather than compile from the resource code.

**Comment by Shauren** [comment]
I don't know. We don't research centos at all and don't test it, so treat it as another reason why we don't recommend it, you are on your own if you insist on using it

**Comment by Aokromes** [comment]
also, centos 7.0 was released 5 years ago, 8.0 this year.
